src/foraging_gui/RigJsonBuilder.py

```diff
@@ -605,8 +602,20 @@ def build_rig_json_core(settings, water_calibration, laser_calibration):
     # Generate Rig Schema
     ###########################################################################
     # Assemble rig schema
+    rig_id = "{}_{}".format(settings['rig_name'],datetime.now().strftime('%Y%m%d'))
+    if re.match(r.RIG_ID_PATTERN, rig_id) is None: # rig_id does not match regex pattern reqs
+        try:  # assuming rigs are named in room-box-letter fashion
+            room, box, letter = settings['rig_name'].split('-')
+            rig_name = room + '_' + box + letter
+            rig_id = "{}_{}".format(rig_name, datetime.now().strftime('%Y%m%d'))
+            if re.match(r.RIG_ID_PATTERN, rig_id) is None: # rig_id still does not match regex pattern reqs
+                raise ValueError
+        except ValueError:
+            logging.error(f'Cannot generate rig because rig_id cannot be configured to match {r.RIG_ID_PATTERN}')
+            return
+
```

Comment:
Can you tell me more about how the rig id pattern changed?

Reply:
The new rig id pattern is 
r"^[a-zA-Z0-9]+_[a-zA-Z0-9-]+_\d{8}$"
so the rig id needs to be formatted like xxxxx_xxxxx_date instead of xxxxx_date. I changed it so now the rig id will look like 446_6A_20240925 which is room_towerbox_date